### PR TITLE
Explicitly reject invalid keys

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/test/integration/remote_cache/remote_cache_test.go [hunk 1]
```diff
@@ -296,6 +296,37 @@ func TestBuild_RemoteCache_ScoreCard(t *testing.T) {
...
+	ctx := context.Background()
+	ws := testbazel.MakeTempWorkspace(t, workspaceContents)
+
+	{
+		// Build 1: explicit empty API key
+		testbazel.Clean(ctx, t, ws)
+		iid := newUUID(t)
+		buildFlags := []string{
+			"//:hello.txt", "--invocation_id=" + iid,
+			"--remote_header=x-buildbuddy-api-key=",
```
Comment (siggisim) [inline enterprise/server/test/integration/remote_cache/remote_cache_test.go:310]: Should we add a test to make sure that non-explicitly set empty keys (i.e. omitted keys) don't fail?

Comment (bduffany) [reply enterprise/server/test/integration/remote_cache/remote_cache_test.go:310]: Good call, this case is tested above, but not intentionally. Done

**Review by tylerwilliams — APPROVED** [review]

File: server/util/prefix/prefix.go [hunk 2]
```diff
@@ -20,9 +21,14 @@ func addPrefix(prefix, key string) string {
...
-		if u, err := auth.AuthenticatedUser(ctx); err == nil {
-			if u.GetGroupID() == "" {
-				return "", status.PermissionDeniedErrorf("Attempting to write to cache as a user with no group.")
+		u, err := auth.AuthenticatedUser(ctx)
+		if err != nil {
+			if authutil.IsInvalidCredentialsError(err) {
+				return "", err
+			}
+		} else {
+			if u.GetGroupID() == "" && !env.GetAuthenticator().AnonymousUsageEnabled() {
```
Comment (vadimberezniker) [inline server/util/prefix/prefix.go:30]: If we get here, doesn't it mean the user is not anonymous? In which case it makes sense to keep the old check, otherwise we will could call addPrefix with an empty group ID.

**Review by vadimberezniker — CHANGES_REQUESTED** [review]

File: server/util/prefix/prefix.go (hunk 2 above)
Comment (bduffany) [reply server/util/prefix/prefix.go]: Reverted back to the old logic in this case.
